src/ipc/interfaces.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <fs.h>
+#include <interfaces/init.h>
+#include <interfaces/ipc.h>
+#include <ipc/capnp/protocol.h>
+#include <ipc/process.h>
+#include <ipc/protocol.h>
+#include <logging.h>
+#include <tinyformat.h>
+#include <util/memory.h>
+#include <util/system.h>
+
+#include <functional>
+#include <memory>
+#include <stdexcept>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <string>
+#include <unistd.h>
+#include <utility>
+#include <vector>
+
+namespace ipc {
+namespace {
+
+class IpcImpl : public interfaces::Ipc
+{
+public:
+    IpcImpl(int argc, char* argv[], const char* exe_name, interfaces::Init& init)
+        : m_protocol(ipc::capnp::MakeCapnpProtocol(exe_name, init)),
+          m_process(ipc::MakeProcess(argc, argv, exe_name, *m_protocol))
+    {
+    }
+    std::unique_ptr<interfaces::Init> spawnProcess(const char* exe_name) override
+    {
+        int pid;
+        int fd = m_process->spawn(exe_name, pid);
+        LogPrint(::BCLog::IPC, "Process %s pid %i launched\n", exe_name, pid);
+        auto init = m_protocol->connect(fd);
+        Ipc::addCleanup(*init, [this, exe_name, pid] {
+            int status = m_process->wait(pid);
+            LogPrint(::BCLog::IPC, "Process %s pid %i exited with status %i\n", exe_name, pid, status);
+        });
+        return init;
+    }
+    bool serveProcess(const char* exe_name, int argc, char* argv[], int& exit_status) override
```

Comment:
I think you can drop `arg` and `argv` parameters, they're already passed in constructors, and those ones should make authority if they differ ?